ryvencore_qt/src/utils.py

```diff
@@ -78,20 +82,36 @@ class MovementEnum(enum.Enum):
     mouse_released = 3
 
 
+def get_resource(filepath: str):
+    # find filepath in xdg data directories
+    path = pathlib.PurePath(filepath)
+    for p in BaseDirectory.load_data_paths(pathlib.PurePath('ryven', path.parent)):
+        if pathlib.Path(p, path.name).is_file():
+            return pathlib.Path(p, path.name)
+
+    # use Location.PACKAGE_PATH
+    return pathlib.Path(Location.PACKAGE_PATH, 'resources', filepath)
+
+
 def change_svg_color(filepath: str, color_hex: str):
     # doesnt seem to work properly yet :(
 
     from qtpy.QtSvg import QSvgRenderer
     from qtpy.QtGui import QPixmap, QPainter
     from qtpy.QtCore import Qt
 
-    with open(filepath) as f:
+    with open(get_resource(filepath)) as f:
         data = f.read()
     data = data.replace('fill:#000000', 'fill:'+color_hex)
-    with open(filepath, 'w') as f:
+
+    path = pathlib.PurePath(filepath)
+    savepath = pathlib.Path(
+        BaseDirectory.save_data_path(
+            pathlib.Path('ryven', path.parent))).with_name(path.name)
+    with open(savepath, 'w') as f:
         f.write(data)
 
-    svg_renderer = QSvgRenderer(filepath)
+    svg_renderer = QSvgRenderer(str(savepath))
     pix = QPixmap(svg_renderer.defaultSize())
     pix.fill(Qt.transparent)
     pix_painter = QPainter(pix)
```

Comment:
do I understand correctly that this enables easily storing (and then using) resources inside xdg's data dir from code, without relying on specific paths? If so, where's the point of doing so if we are never actually storing anything there (at least I can't find it in the code)? The resources of ryvencore-qt are never moved to xdg, no?